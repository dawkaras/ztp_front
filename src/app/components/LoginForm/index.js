import React from 'react';
import { string, func } from 'prop-types';
import Form from 'react-bootstrap/Form';

import Button from 'app/components/Button';

const LoginForm = ({ onSubmit, text }) => {
  return (
    <Form onSubmit={onSubmit} style={{ marginTop: '50px' }}>
      <Form.Group controlId="formBasicEmail">
        <Form.Label>Email:</Form.Label>
        <Form.Control type="text" placeholder="Type your login" />
      </Form.Group>

      <Form.Group controlId="formBasicPassword">
        <Form.Label>Password:</Form.Label>
        <Form.Control type="password" placeholder="Type your password:" />
      </Form.Group>
      <Button text={text} />
    </Form>
  );
};

LoginForm.propTypes = {
  onSubmit: func.isRequired,
  text: string.isRequired,
};
LoginForm.defaultProps = {
  text: '',
};

export default LoginForm;
