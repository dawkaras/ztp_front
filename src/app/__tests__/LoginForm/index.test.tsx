import * as React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from 'app/components/LoginForm';

it('renders LoginForm component', () => {
  const component = renderer
    .create(
      <LoginForm
        onSubmit={() => {
          alert('Test1');
        }}
      />
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
it('renders LoginForm component', () => {
  const component = renderer
    .create(
      <LoginForm
        onSubmit={() => {
          alert('Test2');
        }}
      />
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
