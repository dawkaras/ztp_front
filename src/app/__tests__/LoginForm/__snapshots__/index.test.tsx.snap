// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginForm component 1`] = `
<form
  className=""
  onSubmit={[Function]}
  style={
    Object {
      "marginTop": "50px",
    }
  }
>
  <div>
    <label
      className="form-label"
      htmlFor="formBasicEmail"
    >
      Email:
    </label>
    <input
      className="form-control"
      id="formBasicEmail"
      placeholder="Type your login"
      type="text"
    />
  </div>
  <div>
    <label
      className="form-label"
      htmlFor="formBasicPassword"
    >
      Password:
    </label>
    <input
      className="form-control"
      id="formBasicPassword"
      placeholder="Type your password:"
      type="password"
    />
  </div>
  <button
    className="btn btn-primary"
    color="primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "margin": "5px 0px",
      }
    }
    type="submit"
  >
    
  </button>
</form>
`;

exports[`renders LoginForm component 2`] = `
<form
  className=""
  onSubmit={[Function]}
  style={
    Object {
      "marginTop": "50px",
    }
  }
>
  <div>
    <label
      className="form-label"
      htmlFor="formBasicEmail"
    >
      Email:
    </label>
    <input
      className="form-control"
      id="formBasicEmail"
      placeholder="Type your login"
      type="text"
    />
  </div>
  <div>
    <label
      className="form-label"
      htmlFor="formBasicPassword"
    >
      Password:
    </label>
    <input
      className="form-control"
      id="formBasicPassword"
      placeholder="Type your password:"
      type="password"
    />
  </div>
  <button
    className="btn btn-primary"
    color="primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "margin": "5px 0px",
      }
    }
    type="submit"
  >
    
  </button>
</form>
`;
